src/Hosting/Hosting/src/Internal/WebHost.cs

```diff
@@ -334,7 +334,7 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
 
         public void Dispose()
         {
-            DisposeAsync().GetAwaiter().GetResult();
+            DisposeAsync().AsTask().GetAwaiter().GetResult();
```

Comment:
@stephentoub what are your thoughts on marking `Dispose()` as obsolete where we implement `DisposeAsync()`❔

If it's enough of a win, that would help push people to use the new goodness but it would also be relatively big breaking change when we come to the real removal.

Also, will `ValueTask` no longer implement `GetAwaiter()` soon-ish❔

Reply:
ValueTask will always implement GetAwaiter()... it's what await uses.  When you write:
```C#
await anything;
```
that translates into something along the lines of:
```C#
var awaiter = anything.GetAwaiter();
if (!awaiter.IsCompleted)
{
    _state = 42;
    awaiter.OnCompleted(...);
    return;
}
State42:
awaiter.GetResult();
```
The issue that resulted in this line changing isn't that GetAwaiter doesn't exist but what is the behavior of calling GetResult on it.  A `ValueTask<T>` can wrap one of three things: a T, a `Task<T>`, or an `IValueTaskSource<T>`.  If it's an `IValueTaskSource<T>`, that GetResult call will delegate into whatever that IValueTaskSource implementation is, and pretty much none of them support usage when the operation hasn't yet completed (in contrast to Task, which will block until it's completed). So, you shouldn't use `.GetAwaiter().GetResult()` on a ValueTask unless you know it's already completed.

> what are your thoughts on marking Dispose() as obsolete where we implement DisposeAsync()

I don't think we should, nor do I think it'd help much, since we're not going to mark the interface method as obsolete, and the vast majority usage of Dispose is via IDisposable.